### PR TITLE
Fix problem with open factory page in the selenium tests

### PR DESCRIPTION
**Pull request by Ohrimenko1988**
### What does this PR do?
Resolve problem with redirect to dashboard when try to open factory page by URL

### What issues does this PR fix or reference?
Issue https://github.com/codenvy/codenvy/issues/2477

### CODE REVIEW

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java [hunk 1]
```diff
@@ -48,10 +50,10 @@ public TestFactory(
 
   /** Adds authentication token into the browser and opens factory url. */
   public void authenticateAndOpen(SeleniumWebDriver driver) throws Exception {
-    driver.get(dashboardUrl.get().toString());
-
+    String dashboardUrlPath = dashboardUrl.get().toString();
+    driver.get(dashboardUrlPath);
     entrance.login(owner);
-
+    waitUntilTextPresentInUrl(driver, dashboardUrlPath + "#/");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: "#/" looks like a kind of rudiment

Comment (Ohrimenko1988) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: This is a part which added when dashboard start loading, and exactly this we need check. After that webdriver can open another url without mistakes.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java [hunk 2]
```diff
@@ -48,10 +50,10 @@ public TestFactory(
 
   /** Adds authentication token into the browser and opens factory url. */
   public void authenticateAndOpen(SeleniumWebDriver driver) throws Exception {
-    driver.get(dashboardUrl.get().toString());
-
+    String dashboardUrlPath = dashboardUrl.get().toString();
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: Variable **dashboardUrlPath** looks useless here.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java (hunk 1 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: It's a kind of workaround. Can we wait on presence of some element on page?

Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: We already have login page. IMHO it's more reasonable to check if page has already been loaded inside the **entrance.login()** method.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java [hunk 3]
```diff
@@ -75,4 +77,9 @@ private void deleteFactory() throws Exception {
   private boolean isNamedFactory() {
     return factoryDto.getName() != null;
   }
+
+  private void waitUntilTextPresentInUrl(SeleniumWebDriver driver, String textInUrl) {
+    new WebDriverWait(driver, TestTimeoutsConstants.LOAD_PAGE_TIMEOUT_SEC)
+        .until((WebDriver webdriver) -> webdriver.getCurrentUrl().contains(textInUrl));
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/factory/TestFactory.java]: **startWith()** is more reliable here compare with contains()

**Review by tolusha — APPROVED** [review]

**Review by dmytro-ndp — APPROVED** [review]
Looks good as for me. 
Thank you for quick fixup.

**Pull request merged**